### ISSUE TITLE
forward connector: fails validation across pipelines

### ISSUE THREAD

**Issue by MovieStoreGuy** [open-telemetry/opentelemetry-collector#7892]
**Describe the bug**
it is possible for a shared reference a forward connector to fail validation due it being used across pipelines

**Steps to reproduce**

Have two different telemetry pipelines that share the forward reference, ie:

```yaml
service:
  pipelines:
    metrics/receiving:
      receivers:  [nop]
      processors: []
      exporters:  [forward]
    metrics/exporting:
      receivers:  [forward]
      processors: []
      exporters:  [nop]
    traces/receiving:
      receivers:  [nop]
      processors: []
      exporters:  [forward]
    traces/exporting:
      receivers:  [forward]
      processors: []
      exporters:  [nop]
```

**What did you expect to see?**
This should start up without issue

**What did you see instead?**
An error occurred during the validation step:

```
Error: failed to build pipelines: connector "forward" cannot connect from metrics to logs: telemetry type is not supported
2023/06/14 10:41:50 collector server run finished with error: failed to build pipelines: connector "forward" cannot connect from metrics to logs: telemetry type is not supported
```

**What version did you use?**
Version: v0.79.0

**What config did you use?**
Shared above

**Environment**
OS: docker linux/arm
Compiler: go 1.20.5

**Comment by mx-psi** [comment]
I am putting a p2 priority on this one since we have a workaround (create separate instances of the connector) and it's not (I think) a regression

**Comment by djaglowski** [comment]
Resolved by #8004
